### PR TITLE
ci: replace apm-core-python with python-guild where it makes sense in codeowners

### PR DESCRIPTION
**Pull request by emmettbutler**
This change expands the ownership of a lot of files to the `python-guild` github team, with the goal of widening the bottleneck on pull request approvals. `apm-core-python` is still the default owner for all files, so anything that's overlooked here will still have a backstop reviewer. Hopefully this change reduces the drag felt by contributors to this repository from other Datadog teams.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [ ] Title is accurate
- [ ] All changes are related to the pull request's stated goal
- [ ] Description motivates each change
- [ ] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [ ] Testing strategy adequately addresses listed risks
- [ ] Change is maintainable (easy to change, telemetry, documentation)
- [ ] Release note makes sense to a user of the library
- [ ] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [ ] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -4,12 +4,12 @@
 *                                   @DataDog/apm-core-python
 
 # Framework Integrations
-ddtrace/ext/                        @DataDog/apm-core-python @DataDog/apm-framework-integrations
-ddtrace/contrib/                    @DataDog/apm-core-python @DataDog/apm-framework-integrations
-ddtrace/internal/schema/            @DataDog/apm-core-python @DataDog/apm-framework-integrations
-tests/contrib/                      @DataDog/apm-core-python @DataDog/apm-framework-integrations
-tests/internal/peer_service         @DataDog/apm-core-python @DataDog/apm-framework-integrations
-tests/internal/service_name         @DataDog/apm-core-python @DataDog/apm-framework-integrations
+ddtrace/ext/                        @DataDog/python-guild @DataDog/apm-framework-integrations
```
Comment (brettlangdon) [inline .github/CODEOWNERS:7]: I can't tell if this should be integrations + sdk? or just integrations?

I do see some CI Viz stuff in there, but I think `ddtrace.ext` is meant to be public facing (ish) for tracing constants... sort of...

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -19,12 +19,12 @@ tests/snapshots/                    @DataDog/apm-python
...
-tests/opentelemetry                 @DataDog/apm-core-python
-tests/opentracer                    @DataDog/apm-core-python
-tests/runtime                       @DataDog/apm-core-python
-tests/tracer                        @DataDog/apm-core-python
+tests/internal                      @DataDog/python-guild
+tests/lib-injection                 @DataDog/python-guild
+tests/opentelemetry                 @DataDog/python-guild
+tests/opentracer                    @DataDog/python-guild
+tests/runtime                       @DataDog/python-guild
+tests/tracer                        @DataDog/python-guild
```
Comment (brettlangdon) [inline .github/CODEOWNERS:27]: tracing API team?

**Review by brettlangdon — COMMENTED** [review]
some initial gut reactions

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -4,12 +4,12 @@
...
 # Framework Integrations
-ddtrace/ext/                        @DataDog/apm-core-python @DataDog/apm-framework-integrations
-ddtrace/contrib/                    @DataDog/apm-core-python @DataDog/apm-framework-integrations
-ddtrace/internal/schema/            @DataDog/apm-core-python @DataDog/apm-framework-integrations
-tests/contrib/                      @DataDog/apm-core-python @DataDog/apm-framework-integrations
-tests/internal/peer_service         @DataDog/apm-core-python @DataDog/apm-framework-integrations
-tests/internal/service_name         @DataDog/apm-core-python @DataDog/apm-framework-integrations
+ddtrace/ext/                        @DataDog/python-guild @DataDog/apm-framework-integrations
+ddtrace/contrib/                    @DataDog/python-guild @DataDog/apm-framework-integrations
+ddtrace/internal/schema/            @DataDog/python-guild @DataDog/apm-framework-integrations
```
[the remark this answers is not in the document]
Comment (tabgok) [reply .github/CODEOWNERS:9]: Asking the hard-hitting questions.

Framework-integrations is now wrong, regardless =/

How hard is it to create new github groups?  Can we have "apm-service-naming"? instead?

File: .github/CODEOWNERS (hunk 1 above)
Comment (tabgok) [reply .github/CODEOWNERS:7]: probably ADM + IDM for now

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -89,23 +89,18 @@ tests/contrib/botocore/test_bedrock.py              @DataDog/ml-observability
...
-.github/workflows/system-tests.yml  @DataDog/python-guild  @DataDog/apm-core-python
-ddtrace/internal/compat.py          @DataDog/python-guild  @DataDog/apm-core-python
+.readthedocs.yml                    @DataDog/apm-core-python
+README.md                           @DataDog/apm-core-python
+mypy.ini                            @DataDog/apm-core-python
+.github/PULL_REQUEST_TEMPLATE.md    @DataDog/apm-core-python
+.github/ISSUE_TEMPLATE.md           @DataDog/apm-core-python
+.github/CODEOWNERS                  @DataDog/apm-core-python
+.github/workflows/system-tests.yml  @DataDog/apm-core-python
+ddtrace/internal/compat.py          @DataDog/python-guild
```
[the remark this answers is not in the document]
Comment (emmettbutler) [reply .github/CODEOWNERS:106]: @brettlangdon it looks to me like the 2.7 removal from compat has already happened.

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -20,11 +20,11 @@ tests/snapshots/                    @DataDog/apm-python
...
 
 # Core
-tests/internal                      @DataDog/apm-core-python
+tests/internal                      @DataDog/python-guild
 tests/lib-injection                 @DataDog/apm-core-python
-tests/opentelemetry                 @DataDog/apm-core-python
-tests/opentracer                    @DataDog/apm-core-python
-tests/runtime                       @DataDog/apm-core-python
+tests/opentelemetry                 @DataDog/python-guild
+tests/opentracer                    @DataDog/apm-python
```
Comment (brettlangdon) [inline .github/CODEOWNERS:26]: All of the opentelemetry and opentracer files should be assigned to the API team.

cc @mabdinur who might be able to help us figure out which github team to assign here.

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -89,23 +89,18 @@ tests/contrib/botocore/test_bedrock.py              @DataDog/ml-observability
...
 pyproject.toml                      @DataDog/python-guild
-.readthedocs.yml                    @DataDog/python-guild  @DataDog/apm-core-python
-README.md                           @DataDog/python-guild  @DataDog/apm-core-python
-mypy.ini                            @DataDog/python-guild  @DataDog/apm-core-python
-.github/PULL_REQUEST_TEMPLATE.md    @DataDog/python-guild  @DataDog/apm-core-python
-.github/ISSUE_TEMPLATE.md           @DataDog/python-guild  @DataDog/apm-core-python
-.github/CODEOWNERS                  @DataDog/python-guild  @DataDog/apm-core-python
-.github/workflows/system-tests.yml  @DataDog/python-guild  @DataDog/apm-core-python
-ddtrace/internal/compat.py          @DataDog/python-guild  @DataDog/apm-core-python
+.readthedocs.yml                    @DataDog/apm-core-python
```
[the remark this answers is not in the document]
Comment (brettlangdon) [reply .github/CODEOWNERS:99]: I'm torn, depends on whether we say that language platform is responsible for building/packaging/testing best practices/etc, or whether we delegate that to the guild.

I could be swayed either way.

**Review by brettlangdon — COMMENTED** [review]
We now have @DataDog/lang-platform-python so we probably want to do: `s/apm-core-python/lang-platform-python/g`, wdyt?

**Comment by emmettbutler** [comment]
@brettlangdon I don't think I have the permission level necessary to resolve the linting errors on this pull request. Could you handle that?

**Review by brettlangdon — COMMENTED** [review]
other than getting sdk apis assigned properly, this looks ok to me.

File: .github/CODEOWNERS (hunk 5 above)
Comment (emmettbutler) [reply .github/CODEOWNERS:26]: @brettlangdon @mabdinur what is the "API team" referring to here?

Comment (brettlangdon) [reply .github/CODEOWNERS:26]: @emmettbutler SDK APIs, the new tracing team (the "not IDM" team)

Comment (emmettbutler) [reply .github/CODEOWNERS:26]: @brettlangdon I can't find any team with a name like that in the Github teams list https://github.com/orgs/DataDog/teams?query=api

Comment (brettlangdon) [reply .github/CODEOWNERS:26]: @emmettbutler yeah, they need to create the GitHub team.

**Pull request closed without merging**